packages/core-ethereum/crates/core-ethereum-rpc/src/rpc.rs

```diff
@@ -0,0 +1,326 @@
+use async_trait::async_trait;
+use core_crypto::keypairs::{ChainKeypair, Keypair};
+use core_crypto::types::Hash;
+use core_ethereum_types::{ContractAddresses, ContractInstances};
+use ethers::middleware::{MiddlewareBuilder, NonceManagerMiddleware, SignerMiddleware};
+use ethers::prelude::k256::ecdsa::SigningKey;
+use ethers::prelude::transaction::eip2718::TypedTransaction;
+use ethers::signers::{LocalWallet, Signer, Wallet};
+use ethers::types::BlockId;
+use ethers_providers::{JsonRpcClient, Middleware, Provider, RetryClient, RetryClientBuilder, RetryPolicy};
+use primitive_types::H160;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use std::time::Duration;
+use utils_types::primitives::{Address, Balance, BalanceType, U256};
+use validator::Validate;
+
+use crate::errors::Result;
+use crate::HoprRpcOperations;
+
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Serialize, Deserialize, Validate)]
+pub struct RpcOperationsConfig {
+    pub chain_id: u64,
+    pub contract_addrs: ContractAddresses,
+    pub max_http_retries: u32,
+    pub expected_block_time: Duration,
+    pub logs_page_size: u64,
+    pub tx_polling_interval: Duration,
+}
+
+impl Default for RpcOperationsConfig {
+    fn default() -> Self {
+        Self {
+            chain_id: 100,
```

Comment:
Why have a default for these settings in code instead of our protocol config?

Reply:
We do use the values from protocol config, but we need to have some sensible defaults to satisfy the `Default` trait, but we never use these values.